mutiny-core/src/peermanager.rs

```diff
@@ -321,6 +322,14 @@ pub(crate) async fn connect_peer_if_necessary(
     {
         Ok(())
     } else {
+        // first check to see if the fee rate is mostly up to date
+        // if not, we need to have updated fees or force closures
+        // could occur due to UpdateFee message conflicts.
+        let last_sync = fee_estimator.get_last_sync_time().await;
+        if last_sync.is_none() || utils::now().as_secs() > last_sync.unwrap() + 60 * 10 {
+            fee_estimator.update_fee_estimates().await?;
+        }
```

Comment:
you do this in 2 different places, could just make this a function on the fee estimator like `update_fee_estimates_if_needed`